### DISCUSSION TITLE
What can codespace be used for?

### DISCUSSION THREAD

**Discussion by MrMasrozYTLIVE** [community/community#38788] [Codespaces] [answered]
### Select Topic Area

Question

### Body

What can codespace be used for? Can it be used only for coding or something like hosting simple bots?

**Comment by cmuto09** [comment, marked as answer]
Thanks for the question @MrMasrozYTLIVE!

Codespaces is built for active development, and is not designed for hosting. In fact after a certain amount of time without user activity, a codespace will automatically suspend itself to ensure cost control (see the [idle timeout docs](https://docs.github.com/en/codespaces/managing-codespaces-for-your-organization/restricting-the-idle-timeout-period) for more information)

**Comment by grahamzemel** [comment]
Great question. Codespaces act as in-browser development environments, where you can edit and run programs directly from a larger variety of devices. There's a multitude of programming languages and types of code you can run, so I highly suggest giving it a try. 

They're offered by Github free of charge, and it hibernates if the user has been inactive for too long. That way, it's cost efficient for Github and beneficial to developers. Here's an article with some more info: [Codespaces on Github](https://github.com/features/codespaces).

Hope that helps!
